### PR TITLE
Enhancement/metadata

### LINKED ISSUE #46: Store Totango ID in metadata
Someone messaged me with the following enhancement suggestion (credit where credit is due)
Message below 👇 

regarding [storing metadata in GitHub issues/PRs](https://github.com/jmassardo/issues-to-totango/issues/10#issuecomment-1367068834) you could use this syntax inside the issue comment -

That should hide the data within the Issue but still pass that text (I imagine a key/value pair?) via the API that the app can process. [Here's an example of that](https://github.com/probot/metadata#how-it-works).

### PR DESCRIPTION
**Pull request by rockin-robinson**
This should be merged AFTER Collin's PR for layout updates. 

Closes #46 

Adds task_id and/or touchpoint_id to 'metadata' of issue.

Note: There's no such thing as 'metadata' in GH Issues. This adds an HTML comment that GH strips out when rendering markdown. If someone were to 'edit' the issue body, they would be able to see either 'touchpoint_id' or 'task_id' as an HTML comment.

Testing:
image

### CODE REVIEW

**Comment by srvchawla** [comment]
If we implement Edit functionality via Totango Update API (Tested via curl, just need to implement in node) then we wont be needing the metadata in comments. Thoughts?

**Comment by collinmcneese** [comment]
@rockin-robinson  I noticed a race-condition when I created an issue with both `task` and `touchpoint` labels on it at the same time, both Actions ran but since they fired at the same time only one of the comments persisted in the issue comments since the updates happened at the same time.  Not sure how often this might be seen with both labels at once or a lightweight way to prevent it either, wanted to note it though

**Comment by rockin-robinson** [comment]
> @rockin-robinson I noticed a race-condition when I created an issue with both `task` and `touchpoint` labels on it at the same time, both Actions ran but since they fired at the same time only one of the comments persisted in the issue since the updates happened at the same time. Not sure how often this might be seen with both labels at once or a lightweight way to prevent it either, wanted to note it though

That's interesting because this wouldn't happen if we added a comment (like we were doing) instead of editing the issue body itself. We can talk about during our sync. Thanks!

**Comment by rockin-robinson** [comment]
> @rockin-robinson If we to implement edit/update functionality via Totango Update API (I have tested edit/update via curl, just trying to implement in node) then we wont be needing the metadata in comments. Thoughts?

@srvchawla I'm not familiar with that API. How does it find the Touchpoint/Task in Totango to edit it without an ID?

File: src/totango.js [hunk 1]
```diff
@@ -237,10 +259,11 @@ async function commented({ issue, comment }) {
 
 // Exports for testing
 const totangoPrivate = {
-  comment_gh_issue,
+  add_html_comment,
   create_touchpoint,
   create_task,
   format_body,
+  get_issue_body
```
Comment (sam1el) [inline src/totango.js:266]: you missed a comma here

**Review by sam1el — APPROVED** [review]
Looks solid to me. I like the format.

**Comment by collinmcneese** [comment]
@rockin-robinson I added a minor change on top just to add `npm test` to the build workflow (so we get linting checks) and fixes a minor eslint set of alerts

**Review by collinmcneese — APPROVED** [review]
@rockin-robinson I added a couple things at the end, small
- Added `npm test` to the build workflow so we can catch things on PRs
- Fixed a couple minor eslint errors

**Comment by collinmcneese** [comment]
Aaand I also added a CI workflow for PRs since build is a push event

**Pull request merged**